flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleMaster.java

```diff
@@ -51,6 +52,7 @@
 	 * <p>This call triggers release of any resources which are occupied by the given partition in the external systems
 	 * outside of the producer executor. This is mostly relevant for the batch jobs and blocking result partitions.
 	 * This method is not called if {@link ResultPartitionDeploymentDescriptor#isReleasedOnConsumption()} is {@code true}.
+	 * The partition has to support the {@link ReleaseType#MANUAL} in {@link ShuffleDescriptor#getSupportedReleaseTypes()}.
```

Comment:
In theory different `ShuffleManager` implementations could choose  which release strategy is supported, we should not limit all the implementations must support one type, otherwise we could make this limitation reflected in interface methods.

Reply:
The goal of PR is to let shuffle services decide what to support in `ShuffleDescriptor#getSupportedReleaseTypes`. This doc comment line is related to when the manual release is possible (releasePartitionLocally/releasePartitionExternally).

Or do you mean, that we implicitly assume at the moment that `ReleaseType#MANUAL` has to be always supported for blocking partitions? For this, we have assertion `ResultPartitionDeploymentDescriptor#checkReleaseOnConsumptionIsSupportedForPartition` which will fail the job fast. It is true, we have to make scheduling/release strategies automatically adjust later if something is not supported as written in the comment to the hardcoded `releasedOnConsumption = partitionDescriptor.getPartitionType() != ResultPartitionType.BLOCKING`.

Reply:
Sorry I have not noticed that this javadoc change was for the method of `releasePartitionsExternally`, if so I agree it makes sense that this method implementation is  for MANUL release.
Before I thought the new info was added on the class javadoc, that means all the `ShuffleMaster` implementation must support MANUL release type, so I pointed out that concern.